### PR TITLE
Cleanup/7392 remove softwareproductpage

### LINKED ISSUE #7392: [PNI] Product page - remove SoftwareProduct from the CMS
13-27 September sprint:

- [x] Keep the model around, keep the data, but remove any mention of the SoftwareProduct in the CMS nav/permissible child pages, etc.

Later:

- [ ] clean up the model + data later, once folks have gone over all the old softare products and deemed them removable (if not removed by staff themselves already).

### PR DESCRIPTION
**Pull request by tbrlpld**
To test:

1. Setup project with current production setup: `git switch main`, then `inv copy-stage-db`.
2. Run new migration on the product setup: `git switch cleanup/7392-remove-softwareproductpage`, then `inv migrate`. This should pass without issue. 
3. Start the containers `docker compose up` and visit http://localhost:8000/en/privacynotincluded/. Click around and make sure PNI is still working as expected. Try also visiting it in Wagtail. 
4. Check that a new database also works. <kbd>Control</kbd> + <kbd>C</kbd> then `inv new-db`. This should work without issue.
5. Start the containers again `docker compose up`. Again, visit http://localhost:8000/en/privacynotincluded/ and click around to make sure everything still works as expected. 

***

Closes #7392 
Related PRs/issues #

**Link to sample test page**: https://foundation-s-cleanup-73-alh1lr.herokuapp.com/en/privacynotincluded/
No sample pages. The page type is removed. But, PNI should still be working. 

## Checklist

_Remove unnecessary checks_

**Tests**
- [x] Is the code I'm adding covered by tests? -> No, the tests are removed

**Changes in Models:**
- [x] Did I update or add new fake data? -> No, the fake data is removed
- [x] Did I squash my migration? No, only a single migration file added
- [x] [Are my changes backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team? -> Page usage has already been removed in the past. So this should be backward compatible. 

**Documentation:**
- [x] Is my code documented? -> No
- [x] Did I update the READMEs or wagtail documentation? -> No

### CODE REVIEW

File: network-api/networkapi/wagtailpages/migrations/0009_delete_softwareproductpage.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('wagtailredirects', '0007_add_autocreate_fields'),
+        ('wagtailforms', '0004_add_verbose_name_plural'),
+        ('wagtailinventory', '0002_pageblock_unique_constraint'),
+        ('wagtail_footnotes', '0004_footnote_translatable_mixin_migration'),
+        ('wagtailcore', '0066_collection_management_permissions'),
+        ('wagtailpages', '0008_research_hub_site_structure'),
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/migrations/0009_delete_softwareproductpage.py:14]: very curious why it decided to add all of these when the only dependency is `wagtailpages` O_o

**Comment by Pomax** [comment]
basically R+ with the smallest of changes

File: network-api/networkapi/wagtailpages/migrations/0009_delete_softwareproductpage.py (hunk 1 above)
Comment (tbrlpld) [reply network-api/networkapi/wagtailpages/migrations/0009_delete_softwareproductpage.py:14]: Should I remove the other ones?

**Comment by tbrlpld** [comment]
@Pomax Do you thick a rework of the migrations is still necessary? You have resolved the comment I have seen.

**Comment by Pomax** [comment]
nope. I wish django was smarter here, but in practical terms it doesn't matter.

**Review by Pomax — APPROVED** [review]

**Pull request merged**